datafusion/core/benches/sql_planner.rs

```diff
@@ -115,6 +163,16 @@ fn criterion_benchmark(c: &mut Criterion) {
             )
         })
     });
+
+    let q1_sql = std::fs::read_to_string("../../benchmarks/queries/q1.sql").unwrap();
+    c.bench_function("physical_plan_tpch_q1", |b| {
+        b.iter(|| physical_plan(&ctx, &q1_sql))
+    });
+
+    let q12_sql = std::fs::read_to_string("../../benchmarks/queries/q12.sql").unwrap();
+    c.bench_function("physical_plan_tpch_q12", |b| {
+        b.iter(|| physical_plan(&ctx, &q12_sql))
+    });
```

Comment:
If this is going to add all tpch queries here, I'm wondering if we should create a separate benchmark for tpch?

Reply:
it was my understanding the intent from #8638 was to have all planning benchmarks in a single place and less to be able to isolate benchmark per source (tpch / clickbench / etc).  that being said, i can see value in that if we wanted to take our plan benchmarks a step further and perhaps compare to other engines.  

i dont have a strong opinion either way - happy to go with the consensus here.

Reply:
What do you think about having a benchmark like this that planned them all in a single go. Something like

```rust
    let q1_sql = std::fs::read_to_string("../../benchmarks/queries/q1.sql").unwrap();
    let q12_sql = std::fs::read_to_string("../../benchmarks/queries/q12.sql").unwrap();

    c.bench_function("physical_plan_tpch", |b| {
        b.iter(|| physical_plan(&ctx, &q1_sql))
        b.iter(|| physical_plan(&ctx, &q12_sql))
    });
```

And we can add per-query benchmarks if needed / desired?

I am personally torn between per-query benchmarks which would provide more detail, but require more aggregation to summarize and this single number, with less specificity

Reply:
Indeed, there are several ways we can go about this.  If there isnt an immediate need for per query benchmarks then i think aggregating could work and then we can add per query plans over time or as needed.

Also, and probably not for this PR, it looks like criterion has some features that would allow you to establish a baseline benchmark (https://bheisler.github.io/criterion.rs/book/user_guide/command_line_options.html#baselines).  I'm wondering if that could be useful for integrating into CI to ensure no regressions.

Reply:
There is also the option of using [benchmark_group](https://docs.rs/criterion/latest/criterion/struct.Criterion.html#method.benchmark_group) which allows you to group function benchmarks in final output.